format.md

```diff
@@ -162,11 +162,113 @@ is a list of xy vertices. The child name of the outer list should be "polygons";
+  | 17      | GeometryCollection Z  | `"GeometryCollection Z"`  |
+  | 21      | Point M               | `"Point M"`               |
+  | 22      | LineString M          | `"LineString M"`          |
+  | 23      | Polygon M             | `"Polygon M"`             |
+  | 24      | MultiPoint M          | `"MultiPoint M"`          |
+  | 25      | MultiLineString M     | `"MultiLineString M"`     |
+  | 26      | MultiPolygon M        | `"MultiPolygon M"`        |
+  | 27      | GeometryCollection M  | `"GeometryCollection M"`  |
+  | 31      | Point ZM              | `"Point ZM"`              |
+  | 32      | LineString ZM         | `"LineString ZM"`         |
+  | 33      | Polygon ZM            | `"Polygon ZM"`            |
+  | 34      | MultiPoint ZM         | `"MultiPoint ZM"`         |
+  | 35      | MultiLineString ZM    | `"MultiLineString ZM"`    |
+  | 36      | MultiPolygon ZM       | `"MultiPolygon ZM"`       |
+  | 37      | GeometryCollection ZM | `"GeometryCollection ZM"` |
+
+  These type id values were chosen to match the WKB specification exactly for 2D geometries and match the WKB specification conceptually for Z, M, and ZM geometries, given the constraint that an Arrow union type ID must be between 0 and 127.
+
+- A geometry array does not need to contain _all_ possible children arrays, but the children arrays that it does contain must have the type ids defined above.
+- Children arrays of a geometry array do not need to have the same dimensionality. It will often be easier in practice to store only data of a single dimension, but allowing the geometry array concept to store any mix of types and dimensions enables a static schema for unknown input data.
+- The children arrays should not themselves contain GeoArrow metadata. Only the top-level geometry array should contain GeoArrow metadata.
+- GeometryCollections are allowed as children to make this Geometry array the most general of all geometry types.
+
+    Any geometries stored in GeometryCollection children will not be stored in contiguous memory with other children. E.g. any points within the `GeometryCollection` child will not be contiguous with the `Point` child. If not needed, avoid storing data in GeometryCollection children for best performance.
+
+Note that single and multi geometries of the same type can be stored together in
+a Multi encoding without using this geometry type. For example, a mix of Polygon
+and MultiPolygon can be stored as MultiPolygons, with a Polygon being
+represented as a length-1 MultiPolygon. This is recommended over a geometry
+array if possible because it has less overhead per geometry.
+
+#### GeometryCollection
+
+```
+List<DenseUnion>>
+```
+
+An array of GeometryCollections is represented as a list of a dense union array. In order to explicitly deny support for recursive geometry collections, this definition is not made in terms of the above Geometry array, but re-defines
```

Comment:
Incomplete sentence at the end? 

Also, a nitpick, but since `geometry` union includes GeometryCollection, shouldn't we swap the order of both in the doc?

Reply:
Since Geometry and GeometryCollection are recursive, I don't see any necessity to put either first

Reply:
They are not recursive in both ways? Geometry contains GeometryCollection, but GeometryCollection does not contain Geometry?

Reply:
(I know this is really a stupid nitpick, but it seems to make more sense to first list all the typical geometry sub-types, and then the generic geometry type)